common/src/main/java/com/wynntils/core/notifications/MessageContainer.java

```diff
@@ -9,17 +9,25 @@
 
 public class MessageContainer {
     protected TextRenderTask message;
+    private TextRenderTask currentMessage;
+    private int messageCount;
 
     public MessageContainer(String message) {
-        this.message = new TextRenderTask(message, TextRenderSetting.DEFAULT);
+        this(new TextRenderTask(message, TextRenderSetting.DEFAULT));
     }
 
     public MessageContainer(TextRenderTask message) {
         this.message = message;
+        this.currentMessage = message;
+        this.messageCount = 1;
+    }
+
+    public String getOriginalMessage() {
+        return message.getText();
     }
 
     public TextRenderTask getRenderTask() {
-        return message;
+        return currentMessage;
     }
 
     public void editMessage(String newMessage) {
```

Comment:
Well this won't fly, will it? When do we call this? It will just edit the "original" message, not the one returned by getRenderTask().

*sigh* I was just about to press approve. Are we doing something wrong, design-wise, since this has turned out to be so difficult to get right?

Reply:
I checked. This is called only once, when we pick up emeralds in the pouch in successively in short order.
I guess the best solution is to try to keep this functionality but fix this method. That is, we need to re-create the current message. Which, given the current design, is a bit messy. :(

Reply:
This is fine as is, I think. Yes, it does need to update the cached message, and it does not, but we should save it to the "original" message. I am unsure why we even call it original / current message. Message is the thing we want to display, and we can edit it (for example, `10 emeralds picked up` -> `202 emeralds picked up`). What `currentMessage` currently is just a cached version of the render task that includes the "[x2]", aka the duplication string, so we do not have to allocate it.

Reply:
So, IMO, call `incrementMessageCount` in edit message. Also rename `currentMessage` -> `renderedMessageTaskCache`, as it is no more than that. `getOriginalMessage` should become `getMessage`, as it does not return the __original__ message, but the current version, without the duplication part.